kube-controllers/pkg/controllers/node/ipam.go

```diff
@@ -568,6 +571,26 @@ func (c *ipamController) checkEmptyBlocks() error {
 			continue
 		}
 
+		// Get node to inspect labels
+		node, err := c.client.Nodes().Get(context.TODO(), nodeName, options.GetOptions{})
+		if err != nil {
+			logc.WithError(err).Warn("failed to get node, skipping affinity release")
+			continue
+		}
+
+		// We cannot release blocks affined to nodes undergoing migration from Flannel
+		// Its Pod's IPs are not managed by Calico
+		for labelName, labelVal := range node.ObjectMeta.Labels {
```

Comment:
Would be nice to see this encapsulated in a helper function - I think it simplifies the loop continuation logic as well.

e.g.,

```go
if nodeIsBeingMigrated(node) {
  logc.Info("node is currently undergoing a migration from flannel, do not release blocks")
  continue
}
```

Reply:
If we attach the helper to the ipam controller, it can even include the Get() call above.